Swiften/Elements/ClientState.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta.
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#pragma once
+
+#include <string>
+
+#include <Swiften/Base/API.h>
+#include <Swiften/Elements/Payload.h>
+
+namespace Swift {
+    class SWIFTEN_API ClientState : public Payload {
+        public:
+            typedef std::shared_ptr<ClientState> ref;
+
+        public:
+            enum ClientStateType {Active, Inactive};
+            ClientState(ClientStateType state = Active) {
+                state_ = state;
```

Comment:
This can be move to the initializer list, i.e. `ClientState(ClientStateType state = Active) : state_(state) {`.